pkg/resources/stream_common.go

```diff
@@ -196,24 +201,15 @@ func handleStreamRead(d *schema.ResourceData,
 	stream *sdk.Stream,
 	streamDescription *sdk.Stream,
 ) error {
+	stale, err := booleanStringToBool(*stream.Stale)
```

Comment:
please extract this check with d.Set as a new helper function in resource_helpers_read.go and just use this function in the join below

Reply:
Also, why we are dereferencing the pointer like this here and just not pass a proper value in Stream from SDK? This deref is not safe here.

Reply:
I decided to change it to bool and convert it in the sdk.